src/town_cmd.cpp

```diff
@@ -728,10 +728,17 @@ static CommandCost ClearTile_Town(TileIndex tile, DoCommandFlag flags)
 	Town *t = Town::GetByTile(tile);
 
 	if (Company::IsValidID(_current_company)) {
-		if (rating > t->ratings[_current_company] && !(flags & DC_NO_TEST_TOWN_RATING) &&
-				!_cheats.magic_bulldozer.value && _settings_game.difficulty.town_council_tolerance != TOWN_COUNCIL_PERMISSIVE) {
-			SetDParam(0, t->index);
-			return_cmd_error(STR_ERROR_LOCAL_AUTHORITY_REFUSES_TO_ALLOW_THIS);
+		if (!_cheats.magic_bulldozer.value && !(flags & DC_NO_TEST_TOWN_RATING)) {
+			/* Houses with high enough rating decrease upon removal cannot be removed by the player. */
```

Comment:
```suggestion
			/* NewGRFs can add indestructible houses. */
```

Reply:
In that case we would still have to check the `DC_NO_TEST_TOWN_RATING` flag in `ClearTile_Town()` right? I'm assuming we can't just wrap `if (!CanDeleteHouse(tile)) return CMD_ERROR;` with the flag check either, as the `CanDeleteHouse()` function originally skipped this check if `CBM_HOUSE_DENY_DESTRUCTION` or `BUILDING_IS_PROTECTED` are set.